From 10c2e0610f2006e7644fb36e7c8a70e7ad19d0a9 Mon Sep 17 00:00:00 2001
Message-ID: <10c2e0610f2006e7644fb36e7c8a70e7ad19d0a9.1707833700.git.sharlatanus@gmail.com>
In-Reply-To: <9c3d89f5eb0359dc6493dfc0d24667d2a5c6b484.1707833700.git.sharlatanus@gmail.com>
References: <9c3d89f5eb0359dc6493dfc0d24667d2a5c6b484.1707833700.git.sharlatanus@gmail.com>
From: Felix Gruber <felgru@posteo.net>
Date: Mon, 12 Feb 2024 18:04:11 +0000
Subject: [PATCH v4 4/8] gnu: python-rich: Update to 13.7.0.

* gnu/packages/python-xyz.scm (python-rich): Update to 13.7.0.
[source]: Use git-fetch since pypi package doesn't include tests.
[build-system]: Use pyproject-build-system.
[arguments]: Remove custom 'check phase.
[propagated-inputs]: Remove python-attrs, python-colorama,
python-commonmark and python-typingextensions; add
python-ipywidgets and python-markdown-it-py; replace python-pygments
with python-pygments-next.
[native-inputs]: Add python-poetry-core; remove tests (which are now
included in new [source]).

Change-Id: I15f7928fc376d8c2556ca2965fe44453bd15f248
Signed-off-by: Sharlatan Hellseher <sharlatanus@gmail.com>
---
 gnu/packages/python-xyz.scm | 49 ++++++++++++-------------------------
 1 file changed, 15 insertions(+), 34 deletions(-)

diff --git a/gnu/packages/python-xyz.scm b/gnu/packages/python-xyz.scm
index b3eed56d06..afd5caf541 100644
--- a/gnu/packages/python-xyz.scm
+++ b/gnu/packages/python-xyz.scm
@@ -19192,42 +19192,23 @@ (define-public python-slugid
 (define-public python-rich
   (package
     (name "python-rich")
-    (version "12.4.1")
-    (source (origin
-              (method url-fetch)
-              (uri (pypi-uri "rich" version))
-              (sha256
-               (base32
-                "149vjb4cpf9mz14iig0b6d8065dm8aslp6pc45g9ipmp1wf00ffj"))))
-    (build-system python-build-system)
-    (arguments
-     `(#:phases
-       (modify-phases %standard-phases
-         (replace 'check
-           (lambda* (#:key inputs tests? #:allow-other-keys)
-             (when tests?
-               (copy-recursively (string-append
-                                  (assoc-ref inputs "tests") "/tests")
-                                 "tests")
-               (invoke "python" "-m" "pytest" "-vv")))))))
+    (version "13.7.0")
+    (source
+      (origin
+        ;; There are no tests in the PyPI tarball.
+        (method git-fetch)
+        (uri (git-reference
+              (url "https://github.com/willmcgugan/rich")
+              (commit (string-append "v" version))))
+        (file-name (git-file-name name version))
+        (sha256
+         (base32
+          "0qyhw2vvk17qdmfrmi45z4dd0fkwx3l2qrf3iy0yc2y7pfgrmg9g"))))
+    (build-system pyproject-build-system)
     (propagated-inputs
-     (list python-attrs python-colorama python-commonmark python-pygments
-           python-typing-extensions))
+     (list python-ipywidgets python-markdown-it-py python-pygments-next))
     (native-inputs
-     `(("python-pytest" ,python-pytest)
-       ("tests"
-        ;; The release on pypi comes without tests.  We can't build from this
-        ;; checkout, though, because installation requires an invocation of
-        ;; poetry.
-        ,(origin
-           (method git-fetch)
-           (uri (git-reference
-                 (url "https://github.com/willmcgugan/rich")
-                 (commit (string-append "v" version))))
-           (file-name (git-file-name name version))
-           (sha256
-            (base32
-             "17c3gljn8zv32xnpsgd3fqgqn4r7cdfqri41hridcpbhssdgkyp9"))))))
+     (list python-poetry-core python-pytest))
     (home-page "https://github.com/willmcgugan/rich")
     (synopsis "Render rich text and more to the terminal")
     (description
-- 
2.41.0

